### ISSUE TITLE
how to get the updated response from client PUT

### ISSUE THREAD

**Issue by sazyadav** [badgateway/react-ketting#103]
I need to return the updated state data with PUT response :

const put = async (url: string, data: any) => {
    const client = ClientAPI.getInstance();
    const resource = client.go(url);
    return await resource.put({ data });
    
  };

but getting undefined.
could you please help me to get the updated respone from the PUT request

**Comment by evert** [comment]
Hi @sazyadav . In your case, does the server *change* what was stored and returned the updated body in the HTTP response?

We have an open ticket to support `return-representation` here: https://github.com/badgateway/ketting/issues/22
It would be helpful to know if fixing that ticket solves your problem.

There is a workearound right now though, if your server returns the `Content-Location` header (pointing to itself), this will also work:

```typescript
const put = async (url: string, data: any) => {
  const client = ClientAPI.getInstance();
  const resource = client.go(url);
  await resource.put({ data });
  const newState = await resource.get();
  return newState.data;
};
```

This _will_ be served from cache without doing a HTTP request with your new data if the server returned `Content-Location`. If you didn't return this header, it will return whatever you sent with `put()` last.

**Comment by sazyadav** [comment]
Hi @evert , thanks for your reply.

with  const newState = await resource.get();
  return newState.data;
 Not getting the latest submitted data.
 But with the  js Fetch I am getting:
   const fetchForPut = async <T>(url: string, payload: any): Promise<T> => {
    const requestOptions = {
      method: 'PUT',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    };
    const response = await fetch(url, requestOptions);
    return await response.json();
  }

How can I acheive the same with react ketting.
and also fixinga above mentoined issue ( https://github.com/badgateway/ketting/issues/22 ) would be really helpful for me.

**Comment by evert** [comment]
hi @sazyadav can you edit your question to fix the formatting? it's hard to follow right now. To have syntax highlighting work code blocks need to start with

```
```typescript
```

**Comment by sazyadav** [comment]
```const fetchForPut = async (url: string, payload: any): Promise => {
const requestOptions = {
method: 'PUT',
headers: { 'Content-Type': 'application/json' },
body: JSON.stringify(payload)
};
const response = await fetch(url, requestOptions);
return await response.json();
}```
